File: python/cwe319_transmission.py
import requests

# CWE-319 — Cleartext Transmission of Sensitive Information.


def login(username: str, password: str) -> str:
    # SINK: credentials transmitted over an explicit plaintext http:// scheme.
    url = "http://auth.internal.example.com/login"
    response = requests.post(url, data={"username": username, "password": password})
    return response.text


def fetch_secret(token: str) -> str:
    # SINK: TLS certificate verification disabled while carrying a bearer token.
    return requests.get(
        "https://api.example.com/secret",
        headers={"Authorization": f"Bearer {token}"},
        verify=False,
    ).text

File: js/cwe319Transmission.js
const axios = require('axios');
const https = require('https');

// CWE-319 — Cleartext Transmission of Sensitive Information.

async function login(username, password) {
  // SINK: credentials posted over an explicit plaintext http:// scheme.
  const res = await axios.post('http://auth.internal.example.com/login', {
    username,
    password,
  });
  return res.data;
}

async function fetchSecret(token) {
  // SINK: TLS certificate verification disabled while sending a bearer token.
  const agent = new https.Agent({ rejectUnauthorized: false });
  const res = await axios.get('https://api.example.com/secret', {
    httpsAgent: agent,
    headers: { Authorization: `Bearer ${token}` },
  });
  return res.data;
}

module.exports = { login, fetchSecret };
